pkg/kapp/cmd/appchange/list.go

```diff
@@ -75,14 +97,31 @@ func (t AppChangesTable) Print(ui ui.UI) {
 			uitable.NewHeader("Description"),
 			nsHeader,
 		},
+	}
 
-		SortBy: []uitable.ColumnSort{
+	if t.SortFlag.IsSortByNewestFirst() {
+		table.SortBy = []uitable.ColumnSort{
 			{Column: 1, Asc: false},
-			{Column: 0, Asc: true}, // in case start time are same
-		},
+			{Column: 0, Asc: true},
+		}
+	} else {
+		table.SortBy = []uitable.ColumnSort{
+			{Column: 1, Asc: true},
+			{Column: 0, Asc: true},
+		}
 	}
 
+	isFilterApplied := !t.TimeFlags.BeforeTime.IsZero() || !t.TimeFlags.AfterTime.IsZero()
+
 	for _, change := range t.Changes {
+
+		if isFilterApplied {
```

Comment:
The behaviour would be same even if we remove this `if isFilterApplied` condition, right? Or am I just overthinking 👀

Reply:
Yeah removing this will also do the job. Looks like I was overthinking.

Reply:
+1 looks like a double check
Not sure if we should keep `isFilterApplied` or makes the condition below simpler. 🤔